dropped_metrics_test.go

```diff
@@ -0,0 +1,214 @@
+package veneur_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stripe/veneur/v14"
+	"github.com/stripe/veneur/v14/samplers"
+)
+
+type tcData struct {
+	reason  string
+	metrics []samplers.InterMetric
+}
+
+type tcExpected struct {
+	reason string
+	data   map[string]int64
+}
+
+var testCases = []struct {
+	name     string
+	data     []tcData
+	expected []tcExpected
+}{
+	{
+		name: "simple case",
+		data: []tcData{
+			{
+				reason: "simple",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "simple",
+				data: map[string]int64{
+					"foo": 1,
+				},
+			},
+		},
+	},
+	{
+		name: "simple multiple same metric submitted",
+		data: []tcData{
+			{
+				reason: "simple",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "simple",
+				data: map[string]int64{
+					"foo": 3,
+				},
+			},
+		},
+	},
+	{
+		name: "multiple metrics with different reasons",
+		data: []tcData{
+			{
+				reason: "one",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+				},
+			},
+			{
+				reason: "two",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "bar",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "one",
+				data: map[string]int64{
+					"foo": 1,
+				},
+			},
+			{
+				reason: "two",
+				data: map[string]int64{
+					"bar": 1,
+				},
+			},
+		},
+	},
+	{
+		name: "multiple overlapping metrics with different reasons",
+		data: []tcData{
+			{
+				reason: "one",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "foo",
+					},
+					{
+						Name: "bar",
+					},
+				},
+			},
+			{
+				reason: "two",
+				metrics: []samplers.InterMetric{
+					{
+						Name: "bar",
+					},
+					{
+						Name: "bar",
+					},
+					{
+						Name: "bar",
+					},
+					{
+						Name: "foo",
+					},
+				},
+			},
+		},
+		expected: []tcExpected{
+			{
+				reason: "one",
+				data: map[string]int64{
+					"foo": 3,
+					"bar": 1,
+				},
+			},
+			{
+				reason: "two",
+				data: map[string]int64{
+					"bar": 3,
+					"foo": 1,
+				},
+			},
+		},
+	},
+}
+
+func Test_DroppedMetrics(t *testing.T) {
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			tl, hook := test.NewNullLogger()
+			l := tl.WithField("process", "dropped_metrics_tracker")
+			dmt := veneur.NewDroppedMetricsTracker(l)
+			dmt.Start(context.Background(), time.Second)
+			for _, d := range tc.data {
+				for _, m := range d.metrics {
+					dmt.Track(d.reason, m)
+				}
+			}
+
+			time.Sleep(time.Second * 2)
```

Comment:
avoid `time.Sleep` in unit tests; can we wait on some other thing happening?

Reply:
mm, not without making the rest of the test harder to reason about. We want the "aggregated" result so we want a full "tick" to run before we look at the data coming out of logrus. 

We can mock the ticker using something like https://github.com/benbjohnson/clock to control that behavior but I'm not sure introducing a new package to wrap the native ticker to work around the time.Sleep is worth while.